### PR TITLE
README: rewrite

### LINKED ISSUE #276: Explicitly write down the architecture of TinyKV as a course

### PR DESCRIPTION
**Pull request by tisonkun**
This closes #276 .

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,99 +1,79 @@
...
-Here is a [reading list](doc/reading_list.md) for the knowledge of distributed storage system. Though not all of them are highly related with this course, they can help you construct the knowledge system in this field.
-
-Also, you’d better read the overview of TiKV and PD's design to get a general impression on what you will build:
-
-- TiKV
-  - <https://pingcap.com/blog-cn/tidb-internal-1/> (Chinese Version)
-  - <https://pingcap.com/blog/2017-07-11-tidbinternal1/> (English Version)
-- PD
-  - <https://pingcap.com/blog-cn/tidb-internal-3/> (Chinese Version)
-  - <https://pingcap.com/blog/2017-07-20-tidbinternal3/> (English Version)
```
Comment (Connor1996) [inline README.md]: Should not be removed.

File: README.md [hunk 2]
```diff
@@ -1,99 +1,79 @@
 # The TinyKV Course
 
-This is a series of projects on a key-value storage system built with the Raft consensus algorithm. These projects are inspired by the famous [MIT 6.824](http://nil.csail.mit.edu/6.824/2018/index.html) course but aim to be closer to industry implementations. The whole course is pruned from [TiKV](https://github.com/tikv/tikv) and re-written in Go. After completing this course, you will have the knowledge to implement a horizontally scalable, highly available, key-value storage service with distributed transaction support and a better understanding of TiKV implementation.
```
Comment (Connor1996) [inline README.md:3]: We need to pinpoint the relation between TinyKV and TiKV. One of the intentions of TinyKV is to help students to understand the code of TiKV and cultivate contributors of TiKV.

File: README.md [hunk 3]
```diff
@@ -1,99 +1,79 @@
...
+Clone the source code to your development machine.
 
-Under the binary dir, run the following commands:
-
-```
-mkdir -p data
+```bash
+mkdir -p $GOPATH/src/github.com/tidb-incubator
+cd $GOPATH/src/github.com/tidb-incubator
+git clone https://github.com/tidb-incubator/tinykv.git
```
Comment (Connor1996) [inline README.md]: it's not necessary

File: README.md (hunk 1 above)
Comment (tisonkun) [reply README.md]: Let me try to add the reading list back. We don't display this message at the beginning IMO.

File: README.md (hunk 2 above)
Comment (tisonkun) [reply README.md:3]: OK

File: README.md (hunk 3 above)
Comment (tisonkun) [reply README.md]: Could you elaborate it? I cannot get the point.

File: README.md (hunk 2 above)
Comment (tisonkun) [reply README.md:3]: Updated.

File: README.md (hunk 1 above)
Comment (tisonkun) [reply README.md]: Updated.

File: README.md (hunk 3 above)
Comment (you06) [reply README.md]:
```suggestion
```bash
git clone https://github.com/tidb-incubator/tinykv.git
```

I think @Cornor1996 means this, no need to put it under $GOPATH anymore.

Comment (Connor1996) [reply README.md]: @you06 YES

Comment (tisonkun) [reply README.md]: OK. Let me modify it.

**Comment by tisonkun** [comment]
@Connor1996 @you06 is there any further concern?

File: README.md [hunk 4]
```diff
@@ -1,99 +1,78 @@
...
-- Project2: build a highly available key-value server with Raft
-- Project3: support multi Raft group and balance scheduling on top of Project2
-- Project4: support distributed transaction on top of Project3
+## Course Architecture
 
-**Important note: This course is still under development, and the documentation is incomplete.** Any feedback and contribution is greatly appreciated. Please see help wanted issues if you want to join in the development.
+The whole project is a skeleton code for a key-value server and a scheduler server at the beginning - you need to finish the core logic step by step:
 
-## Course
+* [Standalone KV](doc/project1-StandaloneKV.md)
```
Comment (tisonkun) [inline README.md:11]: @Whorra I read your summary of TinyKV course and think it is wonderful. Would you mind to help reviewing this course architecture? Maybe elaborate a bit on its content.

Comment (JGEthanChen) [reply README.md:11]: Okay. 😊

File: README.md [hunk 5]
```diff
@@ -1,99 +1,78 @@
...
+  * Implement a standalone storage engine.
+  * Implement raw key-value service handlers.
+* [Raft KV](doc/project2-RaftKV.md)
+  * Implement the basic Raft algorithm
+  * Build a fault-tolerant KV server on top of Raft
+  * Add the support of Raft log garbage collection and snapshot
```
Comment (IcePigZDB) [inline README.md]: Maybe there's a `.`missing.

File: README.md [hunk 6]
```diff
@@ -1,99 +1,78 @@
...
+* [Multi-raft KV](doc/project3-MultiRaftKV.md)
+  * Implement membership change and leadership change to Raft algorithm.
+  * Implement conf change and region split on Raft store
```
Comment (IcePigZDB) [inline README.md]: ditto.

File: README.md [hunk 7]
```diff
@@ -1,99 +1,78 @@
...
-- [Project3 - MultiRaftKV](doc/project3-MultiRaftKV.md)
-- [Project4 - Transaction](doc/project4-Transaction.md)
+We provide a [reading list](doc/reading_list.md) for the knowledge of distributed storage system. Though not all of them are highly related with this course, they can help you construct the knowledge system in this field.
 
-## Deploy to a cluster
+Also, you're encouraged to read the overview of TiKV's and PD's design to get a general impression on what you will build:
 
-After you finish the whole implementation, it becomes runnable. You can try TinyKV by deploying it onto a real cluster, and interact with it through TinySQL.
+* TiKV, the design of data storage ([English](https://pingcap.com/blog/2017-07-11-tidbinternal1), [Chinese](https://pingcap.com/blog-cn/tidb-internal-1))
+* PD, the design of scheduling ([English](https://pingcap.com/blog/2017-07-20-tidbinternal3), [Chinese](https://pingcap.com/blog-cn/tidb-internal-3))
```
Comment (IcePigZDB) [inline README.md]: ditto.

File: README.md (hunk 5 above)
Comment (tisonkun) [reply README.md]: Updated. Ditto others

File: README.md [hunk 8]
```diff
@@ -1,99 +1,78 @@
...
-```
-make
-```
+### Prerequisites
 
-It builds the binary of `tinykv-server` and `tinyscheduler-server` to `bin` dir.
+* `git`: The source code of TinyKV is hosted on GitHub as a git repository. To work with git repository, please [install `git`](https://git-scm.com/downloads).
+* `go`: TinyKV is a Go project. To build TinyKV from source, please [install `go`](https://golang.org/doc/install) with version greater or equal to 1.13.
 
-### Run
```
Comment (Connor1996) [inline README.md:73]: I think we can provide the way of running up the cluster, which pinpoints TinyKV can interact with TinySQL

**Review by Connor1996 — COMMENTED** [review]
rest LGTM

File: README.md (hunk 8 above)
Comment (tisonkun) [reply README.md:73]: Let me try to retain the original Run section in a smooth way.

Comment (tisonkun) [reply README.md:73]: Resolved.

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**